x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcArray.java

```diff
@@ -8,49 +8,60 @@
 package org.elasticsearch.xpack.sql.jdbc;
 
 import java.sql.Array;
+import java.sql.JDBCType;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.SQLFeatureNotSupportedException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 
-import static org.elasticsearch.xpack.sql.jdbc.TypeUtils.baseType;
+import static org.elasticsearch.xpack.sql.jdbc.JdbcDatabaseMetaData.columnInfo;
+import static org.elasticsearch.xpack.sql.jdbc.JdbcDatabaseMetaData.memorySet;
 
-public class JdbcArray implements Array {
+class JdbcArray implements Array {
 
-    private final EsType type;
+    public static final String TABLE_NAME = "ARRAY";
+    public static final String COLUMN_INDEX = "index";
+    public static final String COLUMN_VALUE = "value";
+
+    private final JdbcConfiguration cfg;
+    private final EsType baseType;
     private final List<?> values;
+    private boolean freed = false;
 
-    public JdbcArray(EsType type, List<?> values) {
-        this.type = type;
+    JdbcArray(JdbcConfiguration cfg, EsType baseType, List<?> values) {
```

Comment:
I personally don't like that `JdbcArray `constructor needs the cfg now.
What are the actual variables of cfg needed by `memorySet()` internally?
Can you check if we can have another version of memorySet where cfg is not needed?

Can we at least have a second constructor which passed null for cfg?
I'm thinking a bit into the future `JdbcArray` can be used for setting query params to a preparedStmt and then cfg will be needed, when it's actually only for getting it as a `resultSet`.

Reply:
> I personally don't like that JdbcArray constructor needs the cfg now.

Makes sense.

> Can you check if we can have another version of memorySet where cfg is not needed?

We don't. The `JdbcResultSet` constructor itself needs it, but for one thing only: to get a [calendar instance](https://github.com/elastic/elasticsearch/blob/1ee4c50217b56c0f7864ec3ed4b81c5a4d96309a/x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSet.java#L71) (based on the TZ). So a `null` isn't feasible where a ResultSet needs to be built out of an array, unless we relax the `defaultCalendar` to `@Nullable` inside it.

Reply:
Could refactor to have `JdbcResultSet` only take the TZ and a `JdbcArray` just provide a default (`Z`) when not available. But wondering if this change should be made now, or when needed? I'm fine with either.

Reply:
I'd genuinely, prefer to do that now, it's a too much to add the whole cfg.

Reply:
The original intention of passing the entire cfg might have stemmed from the intention to have locales passed in the session configuration too? In any case..

> Can we at least have a second constructor which passed null for cfg?
I'm thinking a bit into the future JdbcArray can be used for setting query params to a preparedStmt

I've updated to now only take the time zone, but no null TZ constructor, since I guess this will be available / needed also for the prepared statement case? OTOH, not sure what a good default would be here - Z or default - for cases the user passes an array of millis which would maybe need to be converted internally by the driver as, for instance, timestamps?